molecule/scenario.py

```diff
@@ -226,8 +232,24 @@ def _setup(self):
             os.makedirs(self.inventory_directory)
 
 
-def ephemeral_directory(path):
+def ephemeral_directory(path=None):
+    """
+    Returns temporary directory to be used by molecule. Morelcule users should
+    not make any assumptions about its location, permissions or its content as
+    this may change in future release.
+    """
     d = os.getenv('MOLECULE_EPHEMERAL_DIRECTORY')
-    if d:
-        return os.path.join(tempfile.gettempdir(), d)
-    return os.path.join(tempfile.gettempdir(), path)
+    if not d:
+        # Darwin is the only known platform that returns user-isolated tempdirs
+        # so we do not need to make them unique to each user.
+        if platform.system() == 'Darwin':
+            d = tempfile.gettempdir()
+        else:
+            d = os.path.join(tempfile.gettempdir(), 'mol.' + getpass.getuser())
```

Comment:
```suggestion
            d = os.path.join(tempfile.gettempdir(), 'molecule.' + getpass.getuser())
```

Might be clearer what's going on or where this folder belongs to.

Reply:
I am afraid that composing a longer path could create issues on some platforms. I am aware of various path lengh limitations on all platforms, so I am inclined to avoid extra length. I was close to pick short folder based on a checksum instead of that approach, which makes it harder to identify.